From cb90cb578827b22bbcb1f58f155490684b8db1da Mon Sep 17 00:00:00 2001
From: Henry Wang <Henry.Wang@arm.com>
Date: Fri, 22 Jul 2022 13:06:30 +0000
Subject: [PATCH] xen/common,arm: Move get_logical_cpu_from_hw_id() to smp.h

Previously, function `get_logical_cpu_from_hw_id()` is private in
`xen/common/boot_cpupools.c`. File `xen/common/boot_cpupools.c` is
highly dependent on device tree, which currently is only used by
Arm. Having said that, this commit moves `get_logical_cpu_from_hw_id()`
to `arm/include/asm/smp.h`, thus enabling others to use this function.

No functional change intended.

Signed-off-by: Henry Wang <Henry.Wang@arm.com>
Upstream-Status: Pending
Issue-Id: SCM-5049
---
 xen/arch/arm/include/asm/smp.h  | 14 ++++++++++++++
 xen/common/sched/boot-cpupool.c | 13 -------------
 2 files changed, 14 insertions(+), 13 deletions(-)

diff --git a/xen/arch/arm/include/asm/smp.h b/xen/arch/arm/include/asm/smp.h
index 4fabdf5310d8..284e660ef8a8 100644
--- a/xen/arch/arm/include/asm/smp.h
+++ b/xen/arch/arm/include/asm/smp.h
@@ -3,6 +3,7 @@
 
 #ifndef __ASSEMBLY__
 #include <xen/cpumask.h>
+#include <xen/init.h>
 #include <asm/current.h>
 #endif
 
@@ -40,6 +41,19 @@ extern unsigned int smp_get_max_cpus(void);
 
 #define cpu_physical_id(cpu) cpu_logical_map(cpu)
 
+static inline int __init get_logical_cpu_from_hw_id(unsigned int hwid)
+{
+    unsigned int i;
+
+    for ( i = 0; i < nr_cpu_ids; i++ )
+    {
+        if ( cpu_physical_id(i) == hwid )
+            return i;
+    }
+
+    return -1;
+}
+
 #endif
 
 /*
diff --git a/xen/common/sched/boot-cpupool.c b/xen/common/sched/boot-cpupool.c
index 5955e6f9a98b..a98680fbb482 100644
--- a/xen/common/sched/boot-cpupool.c
+++ b/xen/common/sched/boot-cpupool.c
@@ -25,19 +25,6 @@ static unsigned int __initdata next_pool_id;
 #define BTCPUPOOLS_DT_WRONG_NODE      (-3)
 #define BTCPUPOOLS_DT_CORRUPTED_NODE  (-4)
 
-static int __init get_logical_cpu_from_hw_id(unsigned int hwid)
-{
-    unsigned int i;
-
-    for ( i = 0; i < nr_cpu_ids; i++ )
-    {
-        if ( cpu_physical_id(i) == hwid )
-            return i;
-    }
-
-    return -1;
-}
-
 static int __init
 get_logical_cpu_from_cpu_node(const struct dt_device_node *cpu_node)
 {
-- 
2.34.1

